x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/DefaultCheckpointProvider.java

```diff
@@ -401,12 +403,12 @@ public void getCheckpointingInfo(
 
         long timestamp = clock.millis();
 
-        getIndexCheckpoints(timeout, ActionListener.wrap(checkpointsByIndex -> {
-            TransformCheckpoint sourceCheckpoint = new TransformCheckpoint(transformConfig.getId(), timestamp, -1L, checkpointsByIndex, 0L);
+        getIndexCheckpoints(timeout, listener.delegateFailure((l, r) -> {
```

Comment:
Would it make sense to keep using the more descriptive name (`checkpointsByIndex`) rather than `r`?